### ISSUE TITLE
[Feature]  Does updating the paymentSheet configuration with users billing and shipping details update payment intent?

### ISSUE THREAD

**Issue by stshelton** [stripe/stripe-ios#2110]
## Is your feature request related to a problem? 
I updated my default billing address and shipping details within the payment sheet configuration but after the purchase is successful (using test cards) shipping and billing address is not updated. I noticed with apple pay updates the paymentIntent's shipping and billing address once user purchase is successful. This would be a nice feature using the payment sheet configuration to update payment intents shipping and billing details. 

## Describe the solution you'd like
Allow payment sheet to updated payment intent with billing and shipping address

## Describe alternatives you've considered
Another way would be to use backend to updated intent but i don't see an option to update billing address within payment intent.

## Additional context
```
    func setUpPaymentSheet(){
        if let stripePaymentSheetData = stripePaymentSheetData{
            STPAPIClient.shared.publishableKey = stripePaymentSheetData.publishableKey
            
            var configuration = PaymentSheet.Configuration()
            configuration.customer = .init(id: stripePaymentSheetData.customer, ephemeralKeySecret: stripePaymentSheetData.ephemeralKey)
            
            
            if let address = getUsersAddress()?.createStripeAddressObject(){
                configuration.shippingDetails = { [weak self] in
                    return  AddressViewController.AddressDetails(address: address)
                }
                
                configuration.defaultBillingDetails.address.line1 = address.line1
                configuration.defaultBillingDetails.address.line2 = address.line2
                configuration.defaultBillingDetails.address.city = address.city
                configuration.defaultBillingDetails.address.postalCode = address.postalCode
                configuration.defaultBillingDetails.address.country = address.country
                
               // configuration.defaultBillingDetails.address = PaymentSheet.ad
            }
            
            configuration.allowsPaymentMethodsRequiringShippingAddress = true

            
            PaymentSheet.FlowController.create(paymentIntentClientSecret: stripePaymentSheetData.paymentIntent, configuration: configuration) { [weak self] result in
              switch result {
              case .failure(let error):
                print(error)
              case .success(let paymentSheetFlowController):
                  self?.paymentSheetFlowController = paymentSheetFlowController
                  self?.paymentMethodLabel = paymentSheetFlowController.paymentOption?.label ?? "Select a payment method"
                  self?.paymentMethodImage = paymentSheetFlowController.paymentOption?.image ?? UIImage(systemName: "plus")!
              }
            }
        }
    }
```

**Comment by yuki-stripe** [comment]
Hello @stshelton,

Today, `defaultBillingDetails` only provides default values for the fields that are displayed in the PaymentSheet. Only the fields that are displayed in the UI end up on the PaymentIntent. We plan to allow you to configure the PaymentSheet to collect full billing address details, if desired.

The shipping address **is** populated on the PaymentIntent when `shippingDetails` is populated, though - what version of the SDK are you using?

Your backend can also update the PaymentIntent - see https://stripe.com/docs/api/payment_intents/update

**Comment by stshelton** [comment]
Im using 23.2.0. Im updating the shipping details within payment sheet configuration. As shown in the above code. I have a breakpoint set for the shippingDetails completion block and it fires once the user clicks `paymentSheet.flowController.confirmbutton`. But my payment intent is not being updated with shipping details, payment intents shipping details is still null. 

Here's my `AddressViewController.AddressDetails(address: address)` object i return in completion block shown in above code. Not sure if its not working because the name and phone number were left nil.
```
(lldb) po AddressViewController.AddressDetails(address: address)
▿ AddressDetails
  ▿ address : Address
    ▿ city : Optional<String>
      - some : "Davie"
    - country : "US"
    - line1 : "3701 SW 117th Ave"
    ▿ line2 : Optional<String>
      - some : ""
    ▿ postalCode : Optional<String>
      - some : "33330"
    ▿ state : Optional<String>
      - some : "3701 SW 117th Ave"
  - name : nil
  - phone : nil
  - isCheckboxSelected : nil
```

**Comment by yuki-stripe** [comment]
Ah, thank you for the extra information. You're correct, `name` is required in order to attach the shipping details to the PaymentIntent (see the docstring for [`PaymentSheet.Configuration.shippingDetails`](https://stripe.dev/stripe-ios/stripe-paymentsheet/Classes/PaymentSheet/Configuration.html#/s:18StripePaymentSheet0bC0C13ConfigurationV15shippingDetailsAA21AddressViewControllerC0gF0VSgycvp))

**Comment by eurias-stripe** [comment]
Hello, this is available as of v23.6.

**Comment by eurias-stripe** [comment]
> Is there any plan to make billing address collection optional for pre iOS 13 SDKs?

Hi @aviya-tabit, we don't usually backport features to older versions of the SDK, so we don't have any plans to do this.
